### PR TITLE
Return user 'uniqueid' with entry UUID

### PR DESCRIPTION
**Pull request by lucasmrod**
- Returns a 'uniqueid' entry on the user dict, with the retrieved UUID.
- If it wasn't able to detect the UUID from the server, then 'uniqueid' will not be a part of the returned dict.

### CODE REVIEW

File: ldap_reader/reader.py [hunk 1]
```diff
@@ -176,6 +183,12 @@ def _build_user_dict(self, result_dict):
             self.config,
             enabled_attrs_dict)
 
+        # Add 'uniqueid'
+        guid_source = self.config.get('dir_guid_source', None)
```
Comment (merickson) [inline ldap_reader/reader.py:184]: We should enforce that `dir_guid_source` exists. Current NetKES usage depends on its existence, as does plans for Semaphor usage. If it doesn't exist, that's an error.

File: ldap_reader/vendor.py [hunk 2]
```diff
@@ -100,3 +103,14 @@ def check_enabled(config, attrs_dict):
     '''
     server_type = config["server_type"]
     return _SERVER_ENABLE_ATTR_MAP[server_type][1](attrs_dict)
+
+
+def fix_guid(config, guid):
```
Comment (merickson) [inline ldap_reader/vendor.py:108]: Plenty good enough structure for now.

Have we verified that RHDS and OpenLDAP don't throw a binary blob at us? This is why the function exists for AD, and our existing RHDS field implementation just specifies `uid` for this field.

Comment (lucasmrod) [inline ldap_reader/vendor.py:108]: I have verified the code against the three servers, and `AD` is the only one that returns the binary blob. 
Regarding RHDS, on my 389-ds setup, `uid` does not work as object UUID field, `nsuniqueid` does.

File: ldap_reader/reader.py (hunk 1 above)
Comment (lucasmrod) [inline ldap_reader/reader.py:184]: Done, it will fail with `KeyError` if not present in the config.

**Pull request merged**